Minor mapper enhancements

ulupo opened the pull request:
#### What does this implement/fix? Explain your changes.
- More precise description of `node_id` in `mapper_quickstart.ipynb`
- Avoid repeated logic between `fit` and `fit_transform` in `Nerve`. To this end, create new hidden method `_graph_data_creation`
- More precision in documenting the `kind` parameter in the mapper cover clases
- Renaming `parallel_clustering_n_jobs` to `n_jobs` and `parallel_clustering_prefer` to `parallel_backend_prefer`

#### Any other comments?
`mapper_quickstart.ipynb` tested on my machine.

Commit:
More precision in documenting kind in mapper covers

Commit:
Avoid repeated logic between fit and fit_transform in Nerve

Commit:
More precise description of node_id in mapper_quickstart

lewtun commented on giotto/mapper/nerve.py:
```diff
@@ -75,11 +75,7 @@ def fit(self, X, y=None):
-        self.X_ = [(node_info[0], *node_info[1])
-                   for node_info in zip(range(len(self.X_)), self.X_)]
-        self.edges_ = list(self._generate_edges(self.X_))
+        self.X_, self.edges_ = self._graph_data_creation(X)
```
I would prefer it if the utility function was called _create_graph_data() since verbs are better for revealing intent

lewtun commented on giotto/mapper/nerve.py:
```diff
@@ -132,6 +124,14 @@ def fit_transform(self, X, y=None, **fit_params):
                 zip(*_X)))
         return graph
 
+    def _graph_data_creation(self, X):
```
Ditto as above

Commit:
Rename parallel_clustering_{n_jobs, prefer}

With n_jobs and parallel_backend_prefer respectively. Also update documentation and mapper_quickstart.ipynb accordingly.
